src/main/java/net/logstash/logback/appender/AbstractLogstashTcpSocketAppender.java

```diff
@@ -252,7 +253,32 @@
      * @see #isGetHostStringPossible()
      */
     private final boolean getHostStringPossible = isGetHostStringPossible();
-    
+
+    /**
+     * The first set destination is random.
+     *
+     * Defaults to disable. The first destination is considered the primary destination.
+     */
+    private boolean randomDestination;
+
+    /**
+     * Object to create the first random index.
+     */
+    private Random firstRandomIndexGenerator = new Random();
+
+    /**
+     * If the size is exceeded, round-robin occurs.
+     * Destination cycles by round-robin.
+     *
+     * When null (the default), disabled round-robin.
+     */
+    private Integer roundRobinSize;
```

Comment:
Ohhhh,  I thought it would just round-robin the next time a connection was needed (i.e. if a connection was broken).  I didn't notice that your change forces a reconnect after every N log messages.  Makes sense now.  If this is the case, I think it would make more sense to use a time based round-robin, rather than message count based.  That way sudden increases of load don't trigger a reconnection storm.

This is kind of similar to the `secondaryConnectionTTL` already implemented.  However, since it applies to any connection.  So, maybe it would just be `connectionTTL`.